ipv8/test/mocking/rest/ipv8.py

```diff
@@ -0,0 +1,63 @@
+import threading
+
+from twisted.internet.task import LoopingCall
+
+from .comunities import TestIdentityCommunity, TestAttestationCommunity
+from ....keyvault.crypto import ECCrypto
+from ....messaging.interfaces.udp.endpoint import UDPEndpoint
+from ....peer import Peer
+from ....peerdiscovery.discovery import RandomWalk
+from ....peerdiscovery.network import Network
+
+
+class TestIPv8(object):
```

Comment:
Ok. I'll try to change it to `MockIPv8`. If that won't work, then I'll change the name of the class to `TestRestIPv8`.

Reply:
I'm not particularly sure if that's possible. Indeed, the classes do share some degree of similarity, but the `TestIPv8` class, written for the REST API, has two significant differences:

* It is not limited to only one overlay (besides the DHT and TrustChain overlays) which is important, since certain endpoints require that the IPv8 object has a number of varying endpoints, in order for them to work. This, I think, increases the versatility of this new IPv8 wrapper and makes it more suitable for testing the REST API. From what I've seen of the other IPv8 wrapper, it is generally used in tests which focus on a particular part of IPv8 (a particular overlay for instance). Endpoints, however, make use of the functionality provided by multiple overlays. That being said, a better way of configuring endpoints is required in this new wrapper. Currently, overlays are hard-coded. I'll work on improving that.  

* The `endpoint` used in the new wrapper is an actual `UDPEndpoint`, which enables HTTP requests to be captured. This time around, peers actually have a local address on which HTTP requests can be received, rather than a bogus one which only complies with the IPv4 format. This should allow the endpoints to be tested against HTTP requests.